client/container_wait_test.go

```diff
@@ -127,3 +128,36 @@ func ExampleClient_ContainerWait_withTimeout() {
 		log.Fatal(err)
 	}
 }
+
+type TestClient_ContainerWait_cancelReader struct{}
+
+func (TestClient_ContainerWait_cancelReader) Read(p []byte) (n int, err error) {
+	return 0, context.Canceled
+}
+
+func TestClient_ContainerWait_canceled(t *testing.T) {
```

Comment:
Perhaps

- rename this test to be similar to the existing tests higher up (e.g. `TestContainerWaitContextCancelled`), so that we can filter tests by common prefixes.
- while doing so, perhaps move this test (and the type) above the `ExampleClient_..` code (which probably should've been in a separate file, but that's for a separate PR)

Reply:
Done.